server/controller/src/main/java/ai/starwhale/mlops/api/DatasetController.java

```diff
@@ -7,60 +7,149 @@
     }
 
     @Override
-    public ResponseEntity<Void> createDatasetVersion(String datasetId, MultipartFile zipFile,
-        String importPath) {
-        return null;
+    public ResponseEntity<ResponseMessage<String>> createDatasetVersion(String projectId, String datasetId,
+        MultipartFile zipFile, String importPath) {
+        UserVO user = userService.currentUser();
+        String versionId = createVersion(projectId, datasetId, zipFile, importPath, user.getId());
+        return ResponseEntity.ok(Code.success
+            .asResponse(String.valueOf(Optional.of(versionId).orElseThrow(ApiOperationException::new))));
     }
 
     @Override
-    public ResponseEntity<Void> modifyDatasetVersionInfo(String datasetId, String versionId,
-        String tag) {
-        return null;
+    public ResponseEntity<ResponseMessage<String>> modifyDatasetVersionInfo(String projectId, String datasetId,
+        String versionId, String tag) {
+        Boolean res = swDatasetService.modifySWMPVersion(
+            Version.builder().id(versionId).tag(tag).build());
+        Assert.isTrue(Optional.of(res).orElseThrow(ApiOperationException::new));
+        return ResponseEntity.ok(Code.success.asResponse("success"));
     }
 
     @Override
-    public ResponseEntity<ResponseMessage<PageInfo<DatasetVO>>> listDataset(Integer pageNum,
-        Integer pageSize) {
-        return null;
+    public ResponseEntity<ResponseMessage<PageInfo<DatasetVO>>> listDataset(String projectId,
+        Integer pageNum, Integer pageSize) {
+        List<DatasetVO> voList = swDatasetService.listSWMP(
+            SWDSObject.builder().projectId(projectId).build(),
+            PageParams.builder().pageNum(pageNum).pageSize(pageSize).build());
+        PageInfo<DatasetVO> pageInfo = new PageInfo<>(voList);
+        return ResponseEntity.ok(Code.success.asResponse(pageInfo));
     }
 
     @Override
-    public ResponseEntity<ResponseMessage<String>> createDataset(String datasetName,
-        MultipartFile zipFile, String importPath) {
-        return null;
+    public ResponseEntity<ResponseMessage<String>> createDataset(String projectId,
+        String datasetName, MultipartFile zipFile, String importPath) {
+        UserVO user = userService.currentUser();
+        String modelId = swDatasetService.addDataset(
+            SWDSObject.builder().projectId(projectId).name(datasetName).ownerId(user.getId())
+                .build());
+        String versionId = createVersion(projectId, modelId, zipFile, importPath, user.getId());
+        return ResponseEntity.ok(Code.success
+            .asResponse(String.valueOf(Optional.of(versionId).orElseThrow(ApiOperationException::new))));
+    }
+
+    private String createVersion(String projectId, String datasetId, MultipartFile zipFile, String importPath, String userId) {
+        String path = importPath;
+        String meta = "";
+        if (zipFile != null) {
```

Comment:
swds will be multipart uploaded later, not a single zip file, it may include some files , like _manifest.yaml,  index.json , data_{}.swds_bin, label_{}.swds_bin.  
![image](https://user-images.githubusercontent.com/590748/160263139-a8fe08a6-3511-48d3-91e0-6874da180d50.png)

mark some `todo` annotations?

Reply:
This api is used for web ui. Is it designed to upload a package?

Reply:
yes